### ISSUE TITLE
[DS-4117] METSRightsCrosswalk doesn't set resource policy type

### ISSUE THREAD

**Issue by dspace-bot** [DSpace/DSpace#7464]
<i>Imported from JIRA <a href="https://jira.lyrasis.org/browse/DS-4117">[DS-4117]</a> created by haoueclf</i>
<p>When a user deposits an item via SWORD by using METSRights metadata  within  the mets file to transfer the embargo period, none resource policy type is set by the <a href="https://github.com/DSpace/DSpace/blob/21fe950593380a5c6dd56ab7a6b0cf6f3e0d23ea/dspace-api/src/main/java/org/dspace/content/crosswalk/METSRightsCrosswalk.java#L414" class="external-link" target="_blank" rel="nofollow noopener">METSRightsCrosswalk class</a>.  This results to the Embargo Date not being displayed to the admin while editing the metadata of such workflowitem in <b>the Upload File(s)</b> (while changing the Embargo Settings of a file) and <b>Verify Submission step</b>.</p>

<p>The provided policies are added by users, therefor the resource policy type: TYPE_CUSTOM should be added.</p>

<p> </p>

**Comment by fribeiro-fccn** [comment]
https://github.com/DSpace/DSpace/pull/2301#issuecomment-2107878206

This still occurs in DSpace 7. The issue is:

I came across this issue and I think I have a similar question regarding the resource policy type. It happens in DSpace 7.6.1 and is related with depositing via Swordv2 (possibily that can occur via other methods too). We need to deposit items with an Embargo and to accomplish we send via Sword a rightsMD in the metx.xml  file (for the bitstreams only) by adding the follwing Section:
```
 <amdSec ID="amd_IDENTIFIER">
    <rightsMD ID="rightsMD_44">
      <mdWrap MDTYPE="OTHER" OTHERMDTYPE="METSRIGHTS">
        <xmlData
          xmlns:rights="http://cosimo.stanford.edu/sdr/metsrights/" xsi:schemaLocation="http://cosimo.stanford.edu/sdr/metsrights/ http://cosimo.stanford.edu/sdr/metsrights.xsd">
          <rights:RightsDeclarationMD
            xmlns:rights="http://cosimo.stanford.edu/sdr/metsrights/" RIGHTSCATEGORY="LICENSED">
            <rights:Context in-effect="false" start-date="2025-01-01" CONTEXTCLASS="GENERAL PUBLIC">
              <rights:Permissions DISCOVER="true" DISPLAY="true" MODIFY="false" DELETE="false" />
            </rights:Context>
          </rights:RightsDeclarationMD>
        </xmlData>
      </mdWrap>
    </rightsMD>
  </amdSec> 
```

The date is added to the bitstream but no TYPE_CUSTOM is filled as can be seen in the picture bellow
![image](https://github.com/DSpace/DSpace/assets/65163266/adf7cb1d-7436-4a4c-b4dd-e17a32a9e360)

It is a bug or are we doing something wrong?

**Comment by tdonohue** [comment]
@fribeiro-fccn : As I noted in [your other comment](https://github.com/DSpace/DSpace/pull/2301#issuecomment-2107878206), this may be related to #9290 which was fixed by #9296 (in 8.x) and #9520 (in 7.x).   I'd recommend trying those fixes to see if it solves the issue.

**Comment by paulo-graca** [comment]
> @fribeiro-fccn : As I noted in [your other comment](https://github.com/DSpace/DSpace/pull/2301#issuecomment-2107878206), this may be related to #9290 which was fixed by #9296 (in 8.x) and #9520 (in 7.x). I'd recommend trying those fixes to see if it solves the issue.

@tdonohue it seems that it addresses the same issue, but only applies to itemImport. In this case the issue is with ingesting items via SWORD and using METS Crosswalks (ingesting via: https://github.com/DSpace/DSpace/blob/dspace-7_x/dspace-api/src/main/java/org/dspace/content/crosswalk/METSRightsCrosswalk.java). The solution was proposed by: https://github.com/DSpace/DSpace/pull/2301 (for DS6), we already test it for DS7 and it works by forcing the `TYPE_CUSTOM`:
```
rp.setRpType(ResourcePolicy.TYPE_CUSTOM);
```

**Comment by tdonohue** [comment]
@paulo-graca or @fribeiro-fccn : If one of you would like to create a new PR with this fix, we can include it in a future release.  It appears the older PR #2301 was approved but auto-closed because it was inactive/obsolete before it could be merged.  A new PR would be necessary.

**Comment by tdonohue** [comment]
Closing. Fixed by #9596
